Source/Core/DolphinQt2/Config/Mapping/IOWindow.h

```diff
@@ -0,0 +1,98 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QDialog>
+#include <QString>
+
+#include "InputCommon/ControllerInterface/Device.h"
+
+class ControlReference;
+class MappingWidget;
+class QAbstractButton;
+class QComboBox;
+class QDialogButtonBox;
+class QListWidget;
+class QVBoxLayout;
+class QPlainTextEdit;
+class QPushButton;
+class QSlider;
+class QSpinBox;
+
+namespace ControllerEmu
+{
+class EmulatedController;
+}
+
+class IOWindow final : public QDialog
+{
+  Q_OBJECT
+public:
+  enum class Type
+  {
+    Input,
+    Output
+  };
+
+  explicit IOWindow(MappingWidget* parent, ControllerEmu::EmulatedController* m_controller,
```

Comment:
In the case of multiple arguments, it depends. For example, if you want to prevent things like so from occurring: [examples](https://ideone.com/7yN37Q) then it should likely be explicit. Constructors without `explicit` on them are considered [converting constructors](http://en.cppreference.com/w/cpp/language/converting_constructor)